### DISCUSSION TITLE
Adaptive MPC

### DISCUSSION THREAD

**Discussion by caxelrud** [franckgaga/ModelPredictiveControl.jl#48] [Ideas]
Is it possible to adapt (modify parameter before an execution) the linear controller model?

**Comment by franckgaga** [comment]
No it is not possible right now. But I'm planning to add this feature. Are you interested in modifying:

1. the plant model,
2. the weights in the objective function, or
3. both ?

Since I'm currently using immutable `struct`, the number of states and the model augmentation of the new plant model would need to be identical to the old one. Would it be too restrictive ?

**Reply by caxelrud** [reply]
Adapting the plant model and tuning (weights in the obj. function) will be great.
It is quite practical to do adaptation and keep the model order the same (number of states).
This last comment should be valid for other features as added integral action, etc.

**Reply by franckgaga** [reply]
Alright I will work on that feature for the next release.

**Reply by franckgaga** [reply]
Adaptive MPC is added in the new version: https://discourse.julialang.org/t/ann-modelpredictivecontrol-jl/100147/27?u=franckgaga
The modification of the weights will be added soon.
Thanks for the constructive comment!

**Comment by franckgaga** [comment]
I added the support for online weights and covariances modification. It should cover all the points here.
